changelog_unreleased/yaml/pr-9674.md

```diff
@@ -0,0 +1,20 @@
+#### Fix `prettier-ignore` after anchors and tags (#9674 by @fisker)
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+- &anchor
+  # prettier-ignore
+  - [1,      2]
+  - [1,      2]
+
+// Prettier stable
+- &anchor # prettier-ignore
+  - [1, 2]
+  - [1, 2]
+
+// Prettier master
+- &anchor # prettier-ignore
+  - [1,      2]
+  - [1,      2]
```

Comment:
I'm not familiar with YAML, so this is just a question. Shouldn't this be like this?
```yaml
- &anchor
  # prettier-ignore
  - [1,      2]
  - [1,      2]
```

Reply:
Yes' it should be, I guess, but it's not the problem this PR fixing. Middle comment format like this

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBaABAMgIZTACwgCcAdKddAYnQFsBLAEwYBs51IaaEYyKMBtAIwAaCmPQAmALq90AkeIrSyIYSAgAHGHWgBnZKGxEiEAO4AFIwn0pszU9gCe+tQCMi2MAGs4MAMoannRQAObIMEQArnBqcDSucEyJADK4IZHYIXAAYsQ02DDaocgg2JEwEKog+DA0zADq+HTwuoFgcH7WzXQAbs2OJWC6LiDBunBEMOYeIfnIAGZ242oAVroAHgBCHt6+fthcycFwC0sxIGvrfsEhrACKkRDwp8zLIIFE40QljgfMVRoiMEYPVGDB8MgABwABjUgIg43qHg0JUBcC+PROagAjo94NNNDZSrpUFA4IlElUiHBcXRqdNMnMkItXudxvRwlE2Td7niTsyzmoYNhXKCGODkBIhR46MwbgBhCCcbAldEAViqkXGABURTYWW8etEAJJQBjcPxgIFaACCZr8MEcrBe4wAvq6gA)
```sh
--parser yaml
```

**Input:**
```yaml
- &anchor
  # middle comment
  - [1,      2]
  - [1,      2]

```

**Output:**
```yaml
- &anchor # middle comment
  - [1, 2]
  - [1, 2]

```